pkg/utils/set/set.go

```diff
@@ -0,0 +1,118 @@
+package set
+
+import (
+	"fmt"
+)
+
+// Set is a generic thread-safe data structure representing a unique collection of items
```

Comment:
Ok, I see your point here, specially after the creation of threadsafe.go. So, we shouldn't call this Set thread-safe. It is actually a thread unsafe type considering it stores the set in a slice, which is thread-unsafe (considering you have methods iterating the slice, or copying it, that might happen in parallel). Despite the semantics of what we call thread-safety, or any discussion of that regareds, what if we just call this "Set" (saying its not thread safe) and using the threadsafe set you created (and even recommending it is always used). Then I would remove the "ItemsUnsafe" method from this also (everything is technically unsafe). Then, if you want to use this unsafe set and synchronize access to it through an external lock/mutex, up to the user (just like any other thread-unsafe type).

Reply:
I'd rather keep the `ItemsUnsafe` method just for the usefulness of a readonly non copying method. Maybe call it `ItemsMutable`?

Reply:
The comment itself I just forgot to change.